apps/src/templates/rubrics/LearningGoals.jsx

```diff
@@ -0,0 +1,392 @@
+  const [errorAutosaving, setErrorAutosaving] = useState(false);
+  const [learningGoalEval, setLearningGoalEval] = useState(null);
+  const [displayFeedback, setDisplayFeedback] = useState('');
+  const [displayUnderstanding, setDisplayUnderstanding] =
+    useState(invalidUnderstanding);
+  const [currentLearningGoal, setCurrentLearningGoal] = useState(0);
+  const teacherFeedback = useRef('');
+  const understandingLevel = useRef(invalidUnderstanding);
+
+  const aiEnabled =
+    learningGoals[currentLearningGoal].aiEnabled && teacherHasEnabledAi;
+  const base_teacher_evaluation_endpoint = '/learning_goal_teacher_evaluations';
+
+  // Timer variables for autosaving
+  const autosaveTimer = useRef();
+  const saveAfter = 2000;
+
+  const handleFeedbackChange = event => {
+    if (studentLevelInfo.user_id && learningGoals[currentLearningGoal].id) {
+      if (autosaveTimer.current) {
+        clearTimeout(autosaveTimer.current);
+      }
+      teacherFeedback.current = event.target.value;
+      setDisplayFeedback(teacherFeedback.current);
+      autosaveTimer.current = setTimeout(() => {
+        autosave();
+      }, saveAfter);
+    }
+  };
+
+  const getAiUnderstanding = learningGoalId => {
+    if (!!aiEvaluations) {
+      const aiInfo = aiEvaluations.find(
+        item => item.learning_goal_id === learningGoalId
+      );
+      return aiInfo?.understanding;
+    } else {
+      return null;
+    }
+  };
+
+  const getAiConfidence = learningGoalId => {
+    if (!!aiEvaluations) {
+      const aiInfo = aiEvaluations.find(
+        item => item.learning_goal_id === learningGoalId
+      );
+      return aiInfo?.ai_confidence;
+    } else {
+      return null;
+    }
+  };
+
+  const getAiInfo = learningGoalId => {
+    if (!!aiEvaluations) {
+      const aiInfo = aiEvaluations.find(
+        item => item.learning_goal_id === learningGoalId
+      );
+      return aiInfo;
+    } else {
+      return null;
+    }
+  };
+
+  const aiUnderstanding = getAiUnderstanding(
+    learningGoals[currentLearningGoal].id
+  );
+  const aiConfidence = getAiConfidence(learningGoals[currentLearningGoal].id);
+  const aiEvalInfo = getAiInfo(learningGoals[currentLearningGoal].id);
+
+  const autosave = () => {
+    setAutosaved(false);
+    setIsAutosaving(true);
+    setErrorAutosaving(false);
+    const bodyData = JSON.stringify({
+      studentId: studentLevelInfo.user_id,
+      learningGoalId: learningGoals[currentLearningGoal].id,
+      feedback: teacherFeedback.current,
+      understanding: understandingLevel.current,
+    });
+    HttpClient.put(
+      `${base_teacher_evaluation_endpoint}/${learningGoalEval.id}`,
+      bodyData,
+      true,
+      {
+        'Content-Type': 'application/json',
+      }
+    )
+      .then(() => {
+        setIsAutosaving(false);
+        setAutosaved(true);
+        if (!feedbackAdded) {
+          setFeedbackAdded(true);
+        }
+      })
+      .catch(error => {
+        console.log(error);
```

Comment:
Is this meant for your own debugging or to ship to users? If the latter maybe a `console.error` would be more appropriate.